# 데이터 확인
### 10차시 이상 사용한 학생 데이터 확인

In [42]:
import pandas as pd

In [43]:
df = pd.read_csv("data.csv")

In [44]:
df.shape

(330, 32)

In [49]:
df

Unnamed: 0  UserID  Sco001  Con001  Pas001  Sco002  Con002  Pas002  \
0           217  210214      63      40      66      59      38      58   
1            69  210271      89      21      96      91      16      96   
2           263  210150      18      82      19       8      99      13   
3            14  210066      75      37      84      84      38     100   
4            18  210274      88      22      96      91      18      97   
..          ...     ...     ...     ...     ...     ...     ...     ...   
325         308  210063      26      91      38      30      59      23   
326          87  210009      83      19      85      75      36      83   
327         250  210314      17      77      14      27      72      27   
328          23  210163      82      33      92      79      26      83   
329         168  210271      64      58      80      63      58      78   

     Sco003  Con003  ...  Pas007  Sco008  Con008  Pas008  Sco009  Con009  \
0        51      56  ...      55      54      56      62      58      37   
1        85      10  ...      85      88      10      88      82      15   
2        14      69  ...      21      36      52      28      18      70   
3        77      27  ...      97      79      35      89      79      25   
4        82      28  ...      96      76      37      86      86      30   
..      ...     ...  ...     ...     ...     ...     ...     ...     ...   
325      26      74  ...      31      27      63      21      38      74   
326      79      37  ...      92      91      16      97      87      12   
327      22      78  ...      25      27      90      39      37      55   
328      84      11  ...      83      86      29      96      92      11   
329      50      60  ...      76      60      60      74      60      60   

     Pas009  Sco010  Con010  Pas010  
0        56      50      50      51  
1        80      82      20      84  
2        11      25      64      18  
3        82      95      11     100  
4        98      77      33      84  
..      ...     ...     ...     ...  
325      46       3      95       3  
326      87      80      30      87  
327      32      28      50      14  
328      95      75      32      81  
329      74      56      35      50  

[330 rows x 32 columns]

# 알고리즘 성능 평가
## 1. 학습 성취도(종합 점수)

- 랜덤으로 생성할 경우, 너무 연관성 없이 데이터가 랜덤하게 생성되기 때문에, 경향을 추가해서 생성하기 위한 코드
- 총 330명 데이터 생성
- 학생 1명씩 순서대로 생성 
- User ID는 반복문 내에서 순서대로 생성 
- 점수 분포는, 상 99명(30%), 중 132명(40%), 하 99명(30%) 분포로 생성

#### 상 99명: Sco 범위 70-100, Con 범위 10-40, Pas 범위 80-100. Sco = 0.4*(100-Con) + 0.6*Pas 실제  72-96
#### 중 132명: Sco 범위 40-70, Con 범위 30-60, Pas 범위 50-80. Sco = 0.4*(100-con) + 0.6*Pas 실제 46-72
#### 하 99명: Sco 범위 0-40, Con 범위 50-100, Pas 범위 0-50. Sco = 0.4*(100-con) + 0.6*Pas 실제 0-50

## 2. 집중도(소요시간)

- 전체 330명 데이터의 90%인 297명의 데이터로 학습을 하여, 나머지 10%인 33명 데이터로 모델의 정확도를 테스트한다.
- 학습은 1번~9번 문제에 대한 집중도(소요시간)를 Input으로 하여, 10번 문제의 집중도(Output)를 예측하는 방식으로 진행한다.

In [55]:
from sklearn.linear_model import LinearRegression
from sklearn.model_selection import train_test_split

In [ ]:
fitter = LinearRegression()
fitter.fit()

In [52]:
# feature = ['UserID', 
#            'Sco001', 'Con001', 'Pas001', 
#            'Sco002', 'Con002', 'Pas002', 
#            'Sco003', 'Con003', 'Pas003', 
#            'Sco004', 'Con004', 'Pas004', 
#            'Sco005', 'Con005', 'Pas005', 
#            'Sco006', 'Con006', 'Pas006', 
#            'Sco007', 'Con007', 'Pas007', 
#            'Sco008', 'Con008', 'Pas008', 
#            'Sco009', 'Con009', 'Pas009', 
#            'Sco010', 'Con010', 'Pas010']
input_feature = ['Con001', 'Con002', 'Con003', 'Con004', 'Con005', 'Con006', 'Con007', 'Con008', 'Con009']
output_feature = ['Con010']

In [53]:
X_train = df[input_feature]

In [ ]:
X_train, X_test, Y_train, Y_test = train_test_split(X,Y)

In [54]:
X_train

Con001  Con002  Con003  Con004  Con005  Con006  Con007  Con008  Con009
0        40      38      56      58      54      32      41      56      37
1        21      16      10      23      39      31      10      10      15
2        82      99      69      77      57      83      90      52      70
3        37      38      27      10      26      31      23      35      25
4        22      18      28      19      11      40      21      37      30
..      ...     ...     ...     ...     ...     ...     ...     ...     ...
325      91      59      74      96      83      75      63      63      74
326      19      36      37      27      16      14      32      16      12
327      77      72      78      61      98      92      95      90      55
328      33      26      11      25      25      11      39      29      11
329      58      58      60      46      31      54      40      60      60

[330 rows x 9 columns]

## 3. 통과 유무(TC 통과율)